lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -778,16 +797,20 @@ def set_mpiexec_options(self):
         # number of procs to start
         if self.options.universe is not None and self.options.universe > 0:
             self.mpiexec_options.append("-np %s" % self.options.universe)
+        elif self.options.hybrid:
+            self.mpiexec_options.append("-np %s" % (len(nub(self.nodes))*self.multiplier))
         else:
-            self.mpiexec_options.append("-np %s" % (self.mpiprocesspernode * len(nub(self.nodes))))
+            self.mpiexec_options.append("-np %s" % (len(self.nodes)*self.multiplier))
 
         # pass local env variables to mpiexec
         self.mpiexec_options += self.get_mpiexec_opts_from_env()
 
     def make_mpiexec_hydra_options(self):
         """Hydra specific mpiexec options."""
         self.get_hydra_info()
-        self.mpiexec_options.append("--hostfile %s" % self.mpiexec_node_filename)
+        # see https://software.intel.com/en-us/articles/controlling-process-placement-with-the-intel-mpi-library
+        # machinefile keeps the imbalance if there is one; hostfile doesn't
```

Comment:
mention `--machinefile` and `--hostfile` to make it clear you're referring to options